proofread: how_to_setup_metadata_engine.md (#1649)
juicedata/juicefs: docs/en/reference/how_to_setup_metadata_engine.md

## Metadata Storage Engine

Metadata is crucially important to a file system as it contains all the detailed information of each file, such as name, size, permissions and location. Especially, for the file system where data and metadata are stored separately, the read and write performance of metadata directly determines the file system performance, and the engine that stores metadata is the most fundamental determinant of performance and reliability.

The metadata storage of JuiceFS uses a multi-engine design. In order to create an ultra-high-performance cloud-native file system, JuiceFS first supports [Redis](https://redis.io), an in-memory Key-Value database, which makes JuiceFS ten times more powerful than Amazon [EFS](https://aws.amazon.com/efs) and [S3FS](https://github.com/s3fs-fuse/s3fs-fuse) performance. Test results can be seen [here](../benchmark/benchmark.md).

> However, we found that many application scenarios do not absolutely rely on high performance based on the feedback from community users. For instance, sometimes users just want to find a convenient tool to migrate data on the cloud or mount the object storage locally to use in a small scale. Therefore, JuiceFS has successively opened up support for more databases such as MySQL/MariaDB and SQLite. The comparison of performance comparison can be found [here](../benchmark/metadata_engines_benchmark.md)).

Author: ```suggestion
However, based on the feedback from community users, we have noticed that a high-performance file system is not urgently required in many application scenarios. Sometimes users just want to find a convenient tool to migrate data on the cloud with high reliability, or to mount the object storage locally to use on a small scale. Therefore, JuiceFS has successively opened up support for more databases such as MySQL/MariaDB and SQLite. The comparison of performance can be found [here](../benchmark/metadata_engines_benchmark.md)).
```

    pics
```

Additional parameters can be appended to the metadata URL. More details can be seen [here](https://pkg.go.dev/github.com/lib/pq#hdr-Connection_String_Parameters).

## MySQL

[MySQL](https://www.mysql.com/) is one of the most popular open source relational databases, and is often preferred for web applications.

### Create a file system

> When using MySQL as the metadata storage engine, you need to create a database manually before create the file system similar to using the databases mentioned above. The command with the following format is usually used to access the database:

Reviewer: ```suggestion
When using MySQL as the metadata storage engine, you need to create a database manually before create the file system. The command with the following format is usually used to access the database:
```
好像原来的那句话就足够说明问题了？😁

```

:::note

> Since BadgerDB is a standalone database, it can only be used locally and does not support multi-host shared mounts. In addition, only one process access is allowed in BadgerDB, and `gc` and `fsck` operations cannot be performed when the file system is mounted.

Reviewer: ```suggestion
Since BadgerDB is a standalone database, it can only be used locally and does not support multi-host shared mounts. In addition, only one process is allowed to access BadgerDB at any time, and `gc` and `fsck` operations cannot be performed when the file system is mounted.
```

> Since BadgerDB is a standalone database, it can only be used locally and does not support multi-host shared mounts. In addition, only one process access is allowed in BadgerDB, and `gc` and `fsck` operations cannot be performed when the file system is mounted.

Author: ```suggestion
Since BadgerDB is a standalone database, it can only be used locally and does not support multi-host shared mounts. In addition, only one process is allowed to access BadgerDB all the time, and `gc` and `fsck` operations cannot be performed when the file system is mounted.
```

:::

## TiKV

[TiKV](https://github.com/tikv/tikv) is a distributed transactional Key-Value database. It is originally developed by [PingCAP](https://pingcap.com) as the storage layer for their flagship product [TiDB](https://github.com/pingcap/tidb). Now TiKV is an independent open source project, and is also a granduated project of [CNCF](https://www.cncf.io/projects).

> By using the official tool `TiUP`, you can easily build a local playground for testing (refer [here](https://tikv.org/docs/5.1/concepts/tikv-in-5-minutes/) for details). Production environment generally requires at least three hosts to store three data replica (refer to the [official document](https://tikv.org/docs/5.1/deploy/install/install/) for all deployment steps).

Reviewer: ```suggestion
By using the official tool `TiUP`, you can easily build a local playground for testing (refer [here](https://tikv.org/docs/5.1/concepts/tikv-in-5-minutes/) for details). Production environment generally requires at least three hosts to store three data replicas (refer to the [official document](https://tikv.org/docs/5.1/deploy/install/install/) for all deployment steps).
```
这里确实应该用复数吧，我看 kubernetes 的 yaml 文件里都是 replicas 😁
